### PR TITLE
feat(cli): suggest matching commands if the user mistypes

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Suggest appropriate commands if the user mistypes utilising the [didyoumean](https://www.npmjs.com/package/didyoumean) package.

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
@sodatea @jkzing Thoughts :thinking:

**Comment by jamesgeorge007** [comment]
@sodatea @jkzing @Akryum Thoughts :thinking:

**Comment by sodatea** [comment]
Thanks for the PR! I like the idea 😀
Will check it later.

**Comment by jamesgeorge007** [comment]
@sodatea Got a chance to check it out :thinking:

File: packages/@vue/cli/bin/vue.js [hunk 1]
```diff
@@ -193,6 +197,7 @@ program
 program
   .arguments('<command>')
   .action((cmd) => {
+    suggestCommands(cmd)
```
Comment (sodatea) [inline packages/@vue/cli/bin/vue.js]: Should print after the help message, otherwise users may not notice it.

File: packages/@vue/cli/bin/vue.js [hunk 2]
```diff
@@ -230,6 +235,17 @@ if (!process.argv.slice(2).length) {
...
 }
 
+function suggestCommands (cmd) {
+  const availableCommands = program.commands.map(cmd => {
+    return cmd._name
+  })
+
+  const suggestion = didYouMean(cmd, availableCommands)
+  if (suggestion) {
+    console.log(chalk.cyan(`Warning: ${chalk.yellow(`Did you mean ${suggestion} ?`)}`))
```
Comment (sodatea) [inline packages/@vue/cli/bin/vue.js]: 1. the space before the question mark needs to be removed.
2. should highlight the command name, as in the `Unknown command …` message. Also the `Warning: ` part seems unnecessary considering the overall coloring.

**Comment by jamesgeorge007** [comment]
@sodatea Thoughts :thinking:

File: packages/@vue/cli/bin/vue.js [hunk 3]
```diff
@@ -242,7 +242,7 @@ function suggestCommands (cmd) {
 
   const suggestion = didYouMean(cmd, availableCommands)
   if (suggestion) {
-    console.log(chalk.cyan(`Warning: ${chalk.yellow(`Did you mean ${suggestion} ?`)}`))
+    console.log(`  ` + chalk.red(`Did you mean ${chalk.yellow(suggestion)}.`))
```
Comment (sodatea) [inline packages/@vue/cli/bin/vue.js]: question mark?

**Comment by jamesgeorge007** [comment]
@sodatea Done :+1:

File: packages/@vue/cli/bin/vue.js [hunk 4]
```diff
@@ -230,6 +235,17 @@ if (!process.argv.slice(2).length) {
...
 }
 
+function suggestCommands (cmd) {
+  const availableCommands = program.commands.map(cmd => {
+    return cmd._name
+  })
+
+  const suggestion = didYouMean(cmd, availableCommands)
+  if (suggestion) {
+    console.log(`  ` + chalk.red(`Did you mean ${chalk.yellow(suggestion)}?.`))
```
Comment (sodatea) [inline packages/@vue/cli/bin/vue.js]: 😂You didn't delete the extra dot here

**Comment by jamesgeorge007** [comment]
@sodatea Hope everythings fine now :sweat_smile:

**Pull request merged**